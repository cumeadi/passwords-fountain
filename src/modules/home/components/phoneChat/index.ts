export { PhoneChat } from './phoneChat.component';
