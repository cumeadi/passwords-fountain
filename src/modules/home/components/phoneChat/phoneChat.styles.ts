import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme, typography, stylelint } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('section')`
    position: relative;
    background: #221e1f;
    border: 5px solid #4b473f;
    min-height: 600px;
    width: 100%;
    max-width: 340px;
    border-radius: 30px;
    overflow: hidden;
`;

export const Screen = styled('div')`
    display: grid;
    margin: 40px 0 5px 5px;
    grid-auto-rows: minmax(min-content, max-content);
    border-radius: 0 0 30px 30px;
    width: calc(100% - 10px);
    padding: ${theme.spacing.s12};
    background: ${theme.colors.white};
`;

const getVariantProperties = ({ isRight }: { isRight: boolean }): string => {
    if (isRight) {
        return stylelint.css`
            justify-self: start;
            background: ${theme.colors.pastelGreen};
        `;
    }

    return stylelint.css`
        justify-self: end;
        background: ${theme.colors.pastelBlue};
    `;
};

export const ChatCloud = styled('div')`
    ${getVariantProperties}
    ${typography.text16}
    width: 90%;
    color: ${theme.colors.darkBlue};
    position: relative;
    border-radius: 15px;
    margin-bottom: 15px;
    padding: 10px;
    line-height: ${theme.lineHeights.s23};

    span {
        color: ${theme.colors.darkBlue};
    }
`;

export const Avatar = styled('img')`
    object-fit: contain;
    position: absolute;
    z-index: ${theme.zIndexes.xs};
    top: 20%;
    ${({ isRight }: { isRight: boolean }) =>
        isRight ? 'right: -34px;' : 'left: -34px;'}
    width: 30px;
    height: 30px;
    border-radius: 50%;
`;

export const Camera = styled('div')`
    position: absolute;
    top: 17px;
    left: 25px;
    height: 10px;
    width: 10px;
    background: #585858;
    border-radius: 10px;
`;

export const CameraLens = styled('div')`
    position: absolute;
    top: 20px;
    left: 28px;
    height: 5px;
    width: 5px;
    background: #0d0d0d;
    border-radius: 5px;
`;

export const Speaker = styled('div')`
    position: absolute;
    top: 17px;
    left: 50%;
    transform: translateX(-50%);
    height: 5px;
    width: 50px;
    background: #574b4d;
    border-radius: 3px;
`;
