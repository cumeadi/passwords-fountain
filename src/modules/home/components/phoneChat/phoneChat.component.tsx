import { h } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import {
    Wrapper,
    Camera,
    CameraLens,
    Speaker,
    Screen,
    ChatCloud,
    Avatar,
} from './phoneChat.styles';
import she from '@/assets/images/she.webp';
import he from '@/assets/images/he.webp';
import { Text } from '@/modules/localisation/components/text';

export const PhoneChat: TypedComponent<Props> = () => {
    return (
        <Wrapper>
            <Camera />
            <CameraLens />
            <Speaker />
            <Screen>
                <ChatCloud>
                    <Text>phoneChat.soAnotherAccount</Text>
                    <Avatar src={she} alt="avatar" />
                </ChatCloud>
                <ChatCloud isRight>
                    <Text>phoneChat.dontYouUse</Text>
                    <Avatar src={he} alt="avatar" isRight />
                </ChatCloud>
                <ChatCloud>
                    <Text>phoneChat.iTried</Text>
                    <Avatar src={she} alt="avatar" />
                </ChatCloud>
                <ChatCloud>
                    <Text>phoneChat.usually</Text>
                    <Avatar src={she} alt="avatar" />
                </ChatCloud>
                <ChatCloud isRight>
                    <Text withMarkup>phoneChat.tryThen</Text>
                    <Avatar src={he} alt="avatar" isRight />
                </ChatCloud>
                <ChatCloud isRight>
                    <Text>phoneChat.itEncodes</Text>
                    <Avatar src={he} alt="avatar" isRight />
                </ChatCloud>
            </Screen>
        </Wrapper>
    );
};

interface Props {}

PhoneChat.propTypes = {};
