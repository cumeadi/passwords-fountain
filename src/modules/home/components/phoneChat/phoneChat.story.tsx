import { h, VNode } from 'preact';
import { PhoneChat } from './phoneChat.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'PhoneChat',
    decorators: [withA11y],
};

export const defaultView = (): VNode => <PhoneChat />;
