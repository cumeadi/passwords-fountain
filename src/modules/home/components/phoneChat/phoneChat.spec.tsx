import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { PhoneChat } from './phoneChat.component';

describe('PhoneChat', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<PhoneChat />);

        expect(asFragment()).toMatchSnapshot();
    });
});
