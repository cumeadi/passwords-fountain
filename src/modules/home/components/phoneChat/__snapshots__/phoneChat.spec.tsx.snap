// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneChat should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go2948064075"
  >
    <div
      class="go1644515284"
    />
    <div
      class="go1680411122"
    />
    <div
      class="go1604521913"
    />
    <div
      class="go34989163"
    >
      <div
        class="go3274727537"
      >
        Oh... so I have to create another account? :Â ( I don't like it.
        <img
          alt="avatar"
          class="go1379153261"
          src="static-asset-stub"
        />
      </div>
      <div
        class="go2924647195"
        isright="true"
      >
        Yes, but don't you use any Password manager?
        <img
          alt="avatar"
          class="go2213353200"
          isright="true"
          src="static-asset-stub"
        />
      </div>
      <div
        class="go3274727537"
      >
        I've tried. I'm not able to trust any of them.
        <img
          alt="avatar"
          class="go1379153261"
          src="static-asset-stub"
        />
      </div>
      <div
        class="go3274727537"
      >
        Usually it's like: you know nothing about where passwords are saved and in which format... and who has access to them?
        <img
          alt="avatar"
          class="go1379153261"
          src="static-asset-stub"
        />
      </div>
      <div
        class="go2924647195"
        isright="true"
      >
        <span
          class="sanitized-translation"
        >
          Try Passwords Fountain then.
          <br />
           It's only lightweight user interface that you use to access your own database.
        </span>
        <img
          alt="avatar"
          class="go2213353200"
          isright="true"
          src="static-asset-stub"
        />
      </div>
      <div
        class="go2924647195"
        isright="true"
      >
        It encrypts all your passwords before sending them + whole code is open source so you be sure that nothing can be stolen.
        <img
          alt="avatar"
          class="go2213353200"
          isright="true"
          src="static-asset-stub"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
